shell/app/common/components/list-select/index.tsx

```diff
@@ -0,0 +1,340 @@
+    setSelectedList((prev) => prev.filter((item) => item[rowKey] !== id));
+  };
+
+  const removeResult = (id: string) => {
+    const currentList = [...resultList];
+    const index = resultList.findIndex((item) => item[rowKey] === id);
+    currentList.splice(index, 1);
+    onChange(currentList);
+    setResultList(currentList);
+  };
+
+  const clear = () => {
+    setSelectedList([]);
+  };
+
+  const onOk = () => {
+    onChange(selectedList);
+    setResultList([...selectedList]);
+    setVisible(false);
+  };
+
+  React.useEffect(() => {
+    value && setResultList([...value]);
+  }, [value]);
+
+  return (
+    <div className="erda-list-select">
+      <Dropdown
+        trigger={['click']}
+        overlay={
+          <ListSelectOverlay
+            {...props}
+            selectedList={selectedList}
+            select={select}
+            remove={remove}
+            onOk={onOk}
+            onCancel={() => setVisible(false)}
+            clear={clear}
+            rowKey={rowKey}
+            menuRowKey={menuRowKey}
+            parentKey={parentKey}
+          />
+        }
+        visible={visible}
+        onVisibleChange={(_visible: boolean) => {
+          _visible && setSelectedList(resultList);
+          setVisible(_visible);
+        }}
+      >
+        <div className="erda-list-select-btn px-2 py-1 rounded-sm inline-flex items-center cursor-pointer">
+          <ErdaIcon type="plus" color="currentColor" size={16} className="mr-1" />
+          {i18n.t('add {name}', { name: label })}
+        </div>
+      </Dropdown>
+      <div className="result-list mt-4 rounded-sm">
+        {resultList.map((item) => (
+          <div
+            className="erda-list-select-selected-item flex items-center p-2 hover:bg-default-04 rounded-sm"
+            key={item[rowKey]}
+          >
+            <div className="flex-1 pr-2">{renderSelectedItem(item)}</div>
+            <ErdaIcon
+              type="close-small"
+              size={24}
+              className="erda-list-select-selected-item-delete cursor-pointer"
+              onClick={() => removeResult(item[rowKey])}
+            />
+          </div>
+        ))}
+      </div>
+    </div>
+  );
+};
+
+interface ListSelectOverlayProps {
+  label: React.ReactNode;
+  multiple?: boolean;
+  menus: Obj[];
+  menusTotal?: number;
+  list: Obj[];
+  onMenuChange?: (item?: Obj) => void;
+  onMenuFilter?: (q: string) => void;
+  onListFilter?: (q: string) => void;
+  onOk?: () => void;
+  onCancel?: () => void;
+  onMenuLoadMore?: (page: number, pageSize?: number) => void;
+  listPagination?: IPagination;
+  renderSelectedItem?: (item: Obj) => React.ReactNode;
+  renderItem?: (item: Obj) => React.ReactNode;
+  select: (item: Obj, checked: boolean) => void;
+  remove: (id: string) => void;
+  clear: () => void;
+  selectedList: Obj[];
+  rowKey: string;
+  menuRowKey: string;
+  parentKey: string;
+  value: Obj[];
+  onChange: (values: Obj[]) => void;
+}
+
+interface IPagination {
```

Comment:
what's the difference between other `IPagination`? maybe u can extend from antd props